feature_engine/feature_selection.py

```diff
@@ -88,3 +89,138 @@ def transform(self, X):
         return X
 
 
+class SelectFeatures(TransformerMixin, BaseEstimator):
+    """
+    SelectFeatures()
+
+    Drops constant and quasi constant variables from a dataframe.
+    Constant variables are those variables which show only one value for all the observations.
+    By default, all constant variables are dropped.
+    Quasi-constant variables are those variables which are almost constant. In other words, those features have the same
+    value for large subset of total observations.
+    This transformer works for both, numerical and categorical variables.
+
+    Parameters
+    ----------
+
+    drop_quasi_constant_features : bool, default=False
+        Whether to drop quasi constant features
+
+    quasi_constant_threshold: float, default=None
+        When drop_quasi_constant_features is True, features showing quasi_constant_threshold percentage of values
+        will be dropped
+
+    Attributes
+    ----------
+
+    constant_features_: list
+        List of constant features in a dataframe
+
+    quasi_constant_features_: list
+        List of quasi constant features in a dataframe
+
+    selected_features_: list
+        List of selected features which are non-constant and non-quasi constant
+
```

Comment:
these 3 attributes are created in the fit() method. We can only create attributes with trailing underscores _ in fit, as the sklearn is_fitted() class looks for these underscores to determine if the transformer is fitted. Please move to fit()